webapp-django/crashstats/crashstats/models.py

```diff
@@ -48,6 +51,8 @@ def config_from_configman():
         definition_source=[
             ElasticsearchConfig.required_config,
             PostgreSQLCrashStorage.required_config,
+            ReprocessingOneRabbitMQCrashStore.required_config,
```

Comment:
I can't get this to work. So much fail. Instead of 
```
implemenation = some_class(config_from_configman())
```
I tried to do 
```
implemenation = some_class(config_from_configman()['database'])
```
but then I get 
```
Traceback (most recent call last):
  File "/opt/boxen/homebrew/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/wsgiref/handlers.py", line 85, in run
    self.result = application(self.environ, self.start_response)
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/django/contrib/staticfiles/handlers.py", line 63, in __call__
    return self.application(environ, start_response)
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/django/core/handlers/wsgi.py", line 189, in __call__
    response = self.get_response(request)
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/django/core/handlers/base.py", line 218, in get_response
    response = self.handle_uncaught_exception(request, resolver, sys.exc_info())
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/peterbe/dev/MOZILLA/SOCORRO/socorro/webapp-django/crashstats/crashstats/decorators.py", line 51, in inner
    versions
  File "/Users/peterbe/dev/MOZILLA/SOCORRO/socorro/webapp-django/crashstats/crashstats/utils.py", line 247, in build_default_context
    for pv in api.get(active=True)['hits']:
  File "/Users/peterbe/dev/MOZILLA/SOCORRO/socorro/webapp-django/crashstats/crashstats/models.py", line 445, in get
    return self._get(expect_json=expect_json, **kwargs)
  File "/Users/peterbe/dev/MOZILLA/SOCORRO/socorro/webapp-django/crashstats/crashstats/models.py", line 486, in _get
    implementation = self.get_implementation()
  File "/Users/peterbe/dev/MOZILLA/SOCORRO/socorro/webapp-django/crashstats/crashstats/models.py", line 400, in get_implementation
    config=config_from_configman()['database']
  File "/Users/peterbe/dev/MOZILLA/SOCORRO/socorro/socorro/external/postgresql/base.py", line 47, in __init__
    self.context
  File "/Users/peterbe/dev/MOZILLA/SOCORRO/socorro/socorro/external/postgresql/crashstorage.py", line 106, in __init__
    quit_check_callback=quit_check_callback
  File "/Users/peterbe/dev/MOZILLA/SOCORRO/socorro/socorro/external/crashstorage_base.py", line 206, in __init__
    self.logger = config.logger
  File "/Users/peterbe/virtualenvs/socorro/lib/python2.7/site-packages/configman/dotdict.py", line 335, in __getattr__
    return getattr(self._parent, key)
ReferenceError: weakly-referenced object no longer exists
```

The defintion_source is set up like this:
```
    definition_source = Namespace()
    definition_source.namespace('es')
    definition_source.es.add_option(
         'es',
         default=ElasticsearchConfig,
    )
    definition_source.namespace('database')
    definition_source.database.add_option(
         'database_class',
         default=PostgreSQLCrashStorage,
    )
    definition_source.namespace('rabbitmq')
    definition_source.rabbitmq.add_option(
         'rabbitmq',
         default=ReprocessingOneRabbitMQCrashStore,
    )
    # definition_source.add_option(
    #     'app',
    #     default=socorro_app.App,
    # )
    return configuration(
        definition_source=[definition_source,socorro_app.App.get_required_config()],
        # definition_source=[
        #     ElasticsearchConfig.get_required_config(),
        #     PostgreSQLCrashStorage.get_required_config(),
        #     ReprocessingOneRabbitMQCrashStore.get_required_config(),
        #
        #     # This required_config defines the logger aggregate
        #     socorro_app.App.get_required_config(),
        # ],
        values_source_list=[
            settings.SOCORRO_IMPLEMENTATIONS_CONFIG,
        ]
    )
```